instrumentation/r2dbc-1.0/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/r2dbc/v1_0/AbstractR2dbcStatementTest.java

```diff
@@ -0,0 +1,272 @@
+                                null,
+                                "SELECT"))),
+                    Arguments.of(
+                        named(
+                            system.system + " Create Table",
+                            new Parameter(
+                                system.system,
+                                "CREATE TABLE person (id SERIAL PRIMARY KEY, first_name VARCHAR(255), last_name VARCHAR(255))",
+                                "CREATE TABLE person (id SERIAL PRIMARY KEY, first_name VARCHAR(?), last_name VARCHAR(?))",
+                                DB,
+                                null,
+                                null))),
+                    Arguments.of(
+                        named(
+                            system.system + " Insert",
+                            new Parameter(
+                                system.system,
+                                "INSERT INTO person (id, first_name, last_name) values (1, 'tom', 'johnson')",
+                                "INSERT INTO person (id, first_name, last_name) values (?, ?, ?)",
+                                "INSERT " + DB + ".person",
+                                "person",
+                                "INSERT"))),
+                    Arguments.of(
+                        named(
+                            system.system + " Select from Table",
+                            new Parameter(
+                                system.system,
+                                "SELECT * FROM person where first_name = 'tom'",
+                                "SELECT * FROM person where first_name = ?",
+                                "SELECT " + DB + ".person",
+                                "person",
+                                "SELECT")))));
+  }
+
+  private static class Parameter {
+
+    public final String system;
+    public final String statement;
+    public final String expectedStatement;
+    public final String spanName;
+    public final String table;
+    public final String operation;
+
+    public Parameter(
+        String system,
+        String statement,
+        String expectedStatement,
+        String spanName,
+        String table,
+        String operation) {
+      this.system = system;
+      this.statement = statement;
+      this.expectedStatement = expectedStatement;
+      this.spanName = spanName;
+      this.table = table;
+      this.operation = operation;
+    }
+  }
+
+  private static class DbSystemProps {
+    public final String system;
+    public final String image;
+    public final int port;
+    public final Map<String, String> envVariables = new HashMap<>();
+
+    public DbSystemProps(String system, String image, int port) {
+      this.system = system;
+      this.image = image;
+      this.port = port;
+    }
+
+    @CanIgnoreReturnValue
+    public DbSystemProps envVariables(String... keyValues) {
+      for (int i = 0; i < keyValues.length - 1; i++) {
```

Comment:
If I interpret this correctly this will end up setting some unintended env variables like `PW_DB=MYSQL_USER`. Perhaps something like
```java
for (int i = 0; i < keyValues.length/2; i++) {
  envVariables.put(keyValues[2*i], keyValues[2*i + 1]);
}
```
would work better?

Reply:
Yes you're right, done.